users_app/tests/test_permissions.py

```diff
@@ -26,3 +29,30 @@ def test_group_creation(self):
 
         for group in groups.keys():
             self.assertTrue(Group.objects.filter(name=group).exists())
+
+
+class TestUserPermissionGroup(TestCase):
```

Comment:
Although your tests make sure users get assigned to the groups you defined, you may want to add assertFalse statements so you make sure they don't get assigned to unexpected groups. As an example, users that qualify for the supplier group shouldn't get assigned any ap group.

Another test that may be useful is to call add_user_to_group with is_new = False, so you make sure users do not get assigned to any new group.

It may not seem necessary now since the code looks straightforward, but it protects your flow from future changes from any other developer. If anyone introduce a breaking change to the flow a test will fail and they will notice.